Code/Framework/AzToolsFramework/AzToolsFramework/UI/Outliner/EntityOutlinerTreeView.cpp

```diff
@@ -228,49 +252,109 @@ namespace AzToolsFramework
             return;
         }
 
-        //if we are attempting to drag an unselected item then we must special case drag and drop logic
-        //QAbstractItemView::startDrag only supports selected items
-        if (m_queuedMouseEvent)
+        // If the index is selected, we should move the whole selection.
+        if (selectionModel()->isSelected(indexAtClick))
         {
-            if (!index.isValid() || index.column() != 0)
-            {
-                return;
-            }
+            StartCustomDrag(selectionModel()->selectedIndexes(), defaultDropAction());
+        }
+        else
+        {
+            StartCustomDrag({ indexAtClick }, defaultDropAction());
+        }
+    }
 
-            if (!selectionModel()->isSelected(index))
-            {
-                StartCustomDrag({ index }, supportedActions);
-                return;
-            }
+    void EntityOutlinerTreeView::SelectAllEntitiesInSelectionRect()
+    {
+        if (!m_queuedMouseEvent)
+        {
+            return;
+        }
+
+        QPoint point1 = (m_queuedMouseEvent->pos());
+        QPoint point2 = (m_mousePosition);
+
+        float top(AZStd::min(point1.y(), point2.y()));
+        float bottom(AZStd::max(point1.y(), point2.y()));
+        float middle(viewport()->rect().center().x());
+
+        QModelIndex topIndex = indexAt(QPoint(middle, top));
+        QModelIndex bottomIndex = indexAt(QPoint(middle, bottom));
+
+        QModelIndex firstIndex = indexAt(QPoint(middle, 10));
+
+        if (!topIndex.isValid() && top < 10)
```

Comment:
update: Would be good to make the `10` a named constant and indicate what it means/is used for